libs/@local/hash-validation/src/data_type.rs

```diff
@@ -0,0 +1,279 @@
+use core::fmt;
+use std::{borrow::Borrow, fmt::Formatter};
```

Comment:
You use `std` here, but then one above `core`, either `no-std` or just `std`. I think it would otherwise get confusing.